### PR TITLE
[PT FE]: support aten::erf and aten::adaptive_avg_pool1d

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *aten::adaptive_avg_pool1d*
 - *aten::erf*
 - *updated aten::adaptive_avg_pool2d and aten::adaptive_max_pool2d* for work with tensors in non-4D format
 - *aten::adaptive_max_pool1d, aten::adaptive_max_pool3d*

### Tickets:
 - *CVS-122486*

### CODE REVIEW

File: src/frontends/pytorch/src/op_table.cpp [hunk 1]
```diff
@@ -229,6 +230,7 @@ const std::map<std::string, CreatorFunction> get_supported_ops_ts() {
         {"aten::acos_", op::inplace_op<op::translate_1to1_match_1_inputs<opset10::Acos>>},
         {"aten::acosh", op::translate_1to1_match_1_inputs_with_fp32_type_alignment<opset10::Acosh>},
         {"aten::acosh_", op::inplace_op<op::translate_1to1_match_1_inputs<opset10::Acosh>>},
+        {"aten::adaptive_avg_pool1d", op::quantizable_op<op::translate_1to1_match_2_inputs<opset10::AdaptiveAvgPool>>},
```
Comment (mvafin) [inline src/frontends/pytorch/src/op_table.cpp:233]: Please make sure it works for non-3D inputs in a same way as torch

File: src/frontends/pytorch/src/op/erf.cpp [hunk 2]
```diff
@@ -17,9 +18,13 @@ OutputVector translate_erf(const NodeContext& context) {
     // aten::erf.out(Tensor self, Tensor(!a) out) -> Tensor(!a)
     num_inputs_check(context, 1, 2);
     auto x = context.get_input(0);
-    
+    auto xdtype = x.get_element_type();
+    if (xdtype.is_dynamic() || !xdtype.is_real()) {
+        x = context.mark_node(std::make_shared<ov::op::v0::Convert>(x, element::f32));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/erf.cpp:23]: Please add comment here, why this is done

File: src/frontends/pytorch/src/op_table.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op_table.cpp]: done

File: src/frontends/pytorch/src/op/erf.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/erf.cpp:24]: done

**Review by mvafin — APPROVED** [review]

**Pull request merged**